It seems like this PR is cursed...

```
2020-01-20T07:31:25.7621459Z [ERROR] Failed to execute goal on project quarkus-kubernetes-deployment: Could not resolve dependencies for project io.quarkus:quarkus-kubernetes-deployment:jar:999-SNAPSHOT: The following artifacts could not be resolved: io.dekorate:dekorate-core:jar:0.10.7, io.dekorate:dekorate-dependencies:jar:0.10.7, io.dekorate:s2i-annotations:jar:0.10.7: Could not find artifact io.dekorate:dekorate-core:jar:0.10.7 in google-maven-central (https://maven-central-eu.storage-download.googleapis.com/repos/central/data/) -> [Help 1]
```

--- a/bom/runtime/pom.xml
+++ b/bom/runtime/pom.xml
@@ -139,7 +139,7 @@
         <awssdk.version>2.10.3</awssdk.version>
         <azure-functions-java-library.version>1.3.0</azure-functions-java-library.version>
         <kotlin.version>1.3.41</kotlin.version>
-        <dekorate.version>0.9.9</dekorate.version>
+        <dekorate.version>0.10.7</dekorate.version>
         <maven-artifact-transfer.version>0.10.0</maven-artifact-transfer.version>
         <jline.version>2.14.6</jline.version>
         <maven-invoker.version>3.0.1</maven-invoker.version>

--- a/integration-tests/kubernetes/src/it/knative/verify.groovy
+++ b/integration-tests/kubernetes/src/it/knative/verify.groovy
@@ -9,9 +9,10 @@ File knativeYml = new File(base, "target/kubernetes/knative.yml")
 assert knativeYml.exists()
 
 //Check that its parse-able
-KubernetesList list = Serialization.unmarshal(knativeYml.text, KubernetesList.class)
-assert list != null
-
+knativeYml.withInputStream { stream ->
+    KubernetesList list = Serialization.unmarshalAsList(stream)
+    assert list != null
+}
 // Un-comment once: https://github.com/fabric8io/kubernetes-client/issues/1838 is addressed.
 //Check that ti contains a DeploymentConfig named after the project
 //Service service = list.items.find{r -> r.kind == "Service"}

--- a/integration-tests/kubernetes/src/it/kubernetes-with-application-properties/verify.groovy
+++ b/integration-tests/kubernetes/src/it/kubernetes-with-application-properties/verify.groovy
@@ -7,31 +7,31 @@ String base = basedir
 File kubernetesYml = new File(base, "target/kubernetes/kubernetes.yml")
 assert kubernetesYml.exists()
 
-//Check that its parse-able
-KubernetesList list = Serialization.unmarshal(kubernetesYml.text, KubernetesList.class)
-assert list != null
+kubernetesYml.withInputStream { stream -> 
+    //Check that its parse-able
+    KubernetesList list = Serialization.unmarshalAsList(stream)
+    assert list != null
+    Deployment deployment = list.items.find{r -> r.kind == "Deployment"}
+    assert deployment != null
+    assert deployment.metadata.name == "test-it"
 
-//Check that ti contains a Deployment named after the project
-Deployment deployment = list.items.find{r -> r.kind == "Deployment"}
-assert deployment != null
-assert deployment.metadata.name == "test-it"
+    //Check that deployment has label foo/bar
+    Map labels = deployment.metadata.labels
+    assert labels.containsKey("foo")
+    assert labels.get("foo") == "bar"
 
-//Check that deployment has label foo/bar
-Map labels = deployment.metadata.labels
-assert labels.containsKey("foo")
-assert labels.get("foo") == "bar"
+    //Check that container has an env var named some value.
+    Container container = deployment.spec.template.spec.containers.get(0)
+    EnvVar env = container.env.find{e -> e.name == "MY_ENV_VAR"}
+    assert env != null
+    assert env.value == "SOMEVALUE"
 
-//Check that container has an env var named some value.
-Container container = deployment.spec.template.spec.containers.get(0)
-EnvVar env = container.env.find{e -> e.name == "MY_ENV_VAR"}
-assert env != null
-assert env.value == "SOMEVALUE"
+    //Check the image
+    assert deployment.spec.template.spec.containers[0].image == "quay.io/grp/kubernetes-with-application-properties:0.1-SNAPSHOT"
+    assert deployment.spec.template.spec.containers[0].ports[0].containerPort == 9090
 
-//Check the image
-assert deployment.spec.template.spec.containers[0].image == "quay.io/grp/kubernetes-with-application-properties:0.1-SNAPSHOT"
-assert deployment.spec.template.spec.containers[0].ports[0].containerPort == 9090
-
-//Check the Service
-Service service = list.items.find{r -> r.kind == "Service"}
-assert service != null
-assert service.spec.ports[0].port == 9090
+    //Check the Service
+    Service service = list.items.find{r -> r.kind == "Service"}
+    assert service != null
+    assert service.spec.ports[0].port == 9090
+}

--- a/integration-tests/kubernetes/src/it/kubernetes-with-client/verify.groovy
+++ b/integration-tests/kubernetes/src/it/kubernetes-with-client/verify.groovy
@@ -8,20 +8,21 @@ import io.dekorate.deps.kubernetes.api.model.rbac.RoleBinding;
 String base = basedir
 File kubernetesYml = new File(base, "target/kubernetes/kubernetes.yml")
 assert kubernetesYml.exists()
+kubernetesYml.withInputStream { stream -> 
+    //Check that its parse-able
+    KubernetesList list = Serialization.unmarshalAsList(stream)
+    assert list != null
+    Deployment deployment = list.items.find{r -> r.kind == "Deployment"}
 
-//Check that its parse-able
-KubernetesList list = Serialization.unmarshal(kubernetesYml.text, KubernetesList.class)
-assert list != null
-Deployment deployment = list.items.find{r -> r.kind == "Deployment"}
+    //Check that ti contains a Deployment named after the project
+    assert deployment != null
+    assert deployment.metadata.name == "kubernetes-with-client"
 
-//Check that ti contains a Deployment named after the project
-assert deployment != null
-assert deployment.metadata.name == "kubernetes-with-client"
+    ServiceAccount serviceAccount = list.items.find{r -> r.kind == "ServiceAccount"}
+    assert serviceAccount != null
+    assert serviceAccount.metadata.name == "kubernetes-with-client"
 
-ServiceAccount serviceAccount = list.items.find{r -> r.kind == "ServiceAccount"}
-assert serviceAccount != null
-assert serviceAccount.metadata.name == "kubernetes-with-client"
-
-RoleBinding rolebinding = list.items.find{r -> r.kind == "RoleBinding"}
-assert rolebinding != null
-assert rolebinding.metadata.name == "kubernetes-with-client:view"
+    RoleBinding rolebinding = list.items.find{r -> r.kind == "RoleBinding"}
+    assert rolebinding != null
+    assert rolebinding.metadata.name == "kubernetes-with-client:view"
+}

--- a/integration-tests/kubernetes/src/it/kubernetes-with-smallrye-health/verify.groovy
+++ b/integration-tests/kubernetes/src/it/kubernetes-with-smallrye-health/verify.groovy
@@ -7,21 +7,22 @@ String base = basedir
 File kubernetesYml = new File(base, "target/kubernetes/kubernetes.yml")
 assert kubernetesYml.exists()
 
-//Check that its parse-able
-KubernetesList list = Serialization.unmarshal(kubernetesYml.text, KubernetesList.class)
-assert list != null
+kubernetesYml.withInputStream { stream -> 
+    //Check that its parse-able
+    KubernetesList list = Serialization.unmarshalAsList(stream)
+    assert list != null
+    Deployment deployment = list.items.find{r -> r.kind == "Deployment"}
 
-//Check that ti contains a Deployment named after the project
-Deployment deployment = list.items.find{r -> r.kind == "Deployment"}
-assert deployment != null
-assert deployment.metadata.name == "kubernetes-with-smallrye-health"
+    assert deployment != null
+    assert deployment.metadata.name == "kubernetes-with-smallrye-health"
 
-//Check that container has probes pointing to smallrye-health endpoint
-Container container = deployment.spec.template.spec.containers.get(0)
-assert container != null
-Probe readiness = container.readinessProbe
-assert readiness != null
-assert readiness.httpGet.path == "/health/ready"
-Probe liveness = container.livenessProbe
-assert liveness != null
-assert liveness.httpGet.path == "/health/live"
+    //Check that container has probes pointing to smallrye-health endpoint
+    Container container = deployment.spec.template.spec.containers.get(0)
+    assert container != null
+    Probe readiness = container.readinessProbe
+    assert readiness != null
+    assert readiness.httpGet.path == "/health/ready"
+    Probe liveness = container.livenessProbe
+    assert liveness != null
+    assert liveness.httpGet.path == "/health/live"
+}

--- a/integration-tests/kubernetes/src/it/kubernetes/verify.groovy
+++ b/integration-tests/kubernetes/src/it/kubernetes/verify.groovy
@@ -7,25 +7,27 @@ import io.dekorate.deps.kubernetes.api.model.Service;
 String base = basedir
 File kubernetesYml = new File(base, "target/kubernetes/kubernetes.yml")
 assert kubernetesYml.exists()
+kubernetesYml.withInputStream { stream -> 
+    //Check that its parse-able
+    KubernetesList list = Serialization.unmarshalAsList(stream)
 
-//Check that its parse-able
-KubernetesList list = Serialization.unmarshal(kubernetesYml.text, KubernetesList.class)
-assert list != null
-Deployment deployment = list.items.find{r -> r.kind == "Deployment"}
-Service service = list.items.find{r -> r.kind == "Service"}
+    assert list != null
+    Deployment deployment = list.items.find{r -> r.kind == "Deployment"}
+    Service service = list.items.find{r -> r.kind == "Service"}
 
-//Check that it contains a Deployment named after the project
-assert deployment != null
-assert deployment.metadata.name == "kubernetes"
+    //Check that it contains a Deployment named after the project
+    assert deployment != null
+    assert deployment.metadata.name == "kubernetes"
 
-//Check the port of the Deployment
-assert deployment.spec.template.spec.containers[0].ports[0].containerPort == 8080
+    //Check the port of the Deployment
+    assert deployment.spec.template.spec.containers[0].ports[0].containerPort == 8080
 
-//Check that it contains a Service named after the project
-assert service != null
-assert service.metadata.name == "kubernetes"
+    //Check that it contains a Service named after the project
+    assert service != null
+    assert service.metadata.name == "kubernetes"
 
-//Check the port of the Service
-assert service.spec.ports[0].port == 8080
+    //Check the port of the Service
+    assert service.spec.ports[0].port == 8080
 
 
+}

--- a/integration-tests/kubernetes/src/it/openshift-with-application-properties/verify.groovy
+++ b/integration-tests/kubernetes/src/it/openshift-with-application-properties/verify.groovy
@@ -4,30 +4,32 @@ import io.dekorate.deps.openshift.api.model.DeploymentConfig;
 
 //Check that file exits
 String base = basedir
-File kubernetesYml = new File(base, "target/kubernetes/openshift.yml")
-assert kubernetesYml.exists()
+File openshiftYml = new File(base, "target/kubernetes/openshift.yml")
+assert openshiftYml.exists()
 
-//Check that its parse-able
-KubernetesList list = Serialization.unmarshal(kubernetesYml.text, KubernetesList.class)
-assert list != null
+openshiftYml.withInputStream { stream ->
+    //Check that its parse-able
+    KubernetesList list = Serialization.unmarshalAsList(stream)
+    assert list != null
 
-//Check that ti contains a Deployment named after the project
-DeploymentConfig deployment = list.items.find{r -> r.kind == "DeploymentConfig"}
-assert deployment != null
-assert deployment.metadata.name == "test-it"
+    //Check that ti contains a Deployment named after the project
+    DeploymentConfig deployment = list.items.find{r -> r.kind == "DeploymentConfig"}
+    assert deployment != null
+    assert deployment.metadata.name == "test-it"
 
-//Check that deployment has label foo/bar
-Map labels = deployment.metadata.labels
-assert labels.containsKey("foo")
-assert labels.get("foo") == "bar"
+    //Check that deployment has label foo/bar
+    Map labels = deployment.metadata.labels
+    assert labels.containsKey("foo")
+    assert labels.get("foo") == "bar"
 
-//Check that container has an env var named some value.
-Container container = deployment.spec.template.spec.containers.get(0)
-EnvVar env = container.env.find{e -> e.name == "MY_ENV_VAR"}
-assert env != null
-assert env.value == "SOMEVALUE"
+    //Check that container has an env var named some value.
+    Container container = deployment.spec.template.spec.containers.get(0)
+    EnvVar env = container.env.find{e -> e.name == "MY_ENV_VAR"}
+    assert env != null
+    assert env.value == "SOMEVALUE"
 
-//Check the Service
-Service service = list.items.find{r -> r.kind == "Service"}
-assert service != null
-assert service.spec.ports[0].port == 9090
+    //Check the Service
+    Service service = list.items.find{r -> r.kind == "Service"}
+    assert service != null
+    assert service.spec.ports[0].port == 9090
+}

--- a/integration-tests/kubernetes/src/it/openshift/verify.groovy
+++ b/integration-tests/kubernetes/src/it/openshift/verify.groovy
@@ -7,13 +7,15 @@ String base = basedir
 File openshiftYml = new File(base, "target/kubernetes/openshift.yml")
 assert openshiftYml.exists()
 
-//Check that its parse-able
-KubernetesList list = Serialization.unmarshal(openshiftYml.text, KubernetesList.class)
-assert list != null
+openshiftYml.withInputStream { stream ->
+    //Check that its parse-able
+    KubernetesList list = Serialization.unmarshalAsList(stream)
+    assert list != null
 
-//Check that ti contains a DeploymentConfig named after the project
-DeploymentConfig deployment = list.items.find{r -> r.kind == "DeploymentConfig"}
-assert deployment != null
+    //Check that ti contains a DeploymentConfig named after the project
+    DeploymentConfig deployment = list.items.find{r -> r.kind == "DeploymentConfig"}
+    assert deployment != null
+}
 
 Container container = deployment.spec.template.spec.containers[0]
 assert container != null

--- a/integration-tests/kubernetes/src/it/openshift/verify.groovy
+++ b/integration-tests/kubernetes/src/it/openshift/verify.groovy
@@ -15,11 +15,12 @@ openshiftYml.withInputStream { stream ->
     //Check that ti contains a DeploymentConfig named after the project
     DeploymentConfig deployment = list.items.find{r -> r.kind == "DeploymentConfig"}
     assert deployment != null
+
+    Container container = deployment.spec.template.spec.containers[0]
+    assert container != null
+
+    EnvVar envVar = container.env.find{e -> e.name == "JAVA_APP_JAR"}
+    assert envVar != null
+    assert envVar.value.endsWith("-runner.jar")
 }
 
-Container container = deployment.spec.template.spec.containers[0]
-assert container != null
-
-EnvVar envVar = container.env.find{e -> e.name == "JAVA_APP_JAR"}
-assert envVar != null
-assert envVar.value.endsWith("-runner.jar")

